packages/core/js-client/src/services/SingleModuleSrv.ts

```diff
@@ -53,7 +51,11 @@ export class Srv {
 
     try {
       const newServiceId = uuidv4();
-      const buffer = Buffer.from(wasmContent, "base64");
+
+      const buffer = Uint8Array.from(atob(wasmContent), (m) => {
+        return m.codePointAt(0) ?? 0;
+      });
```

Comment:
are you sure this operation is correct and safe? Looks a bit sketchy, I would at least move it to a separate function and give it a readable name so at least it's easier to understand what is happening really. Also you have several variables that are named contain the word `buffer`. Maybe rename it to something more useful (and it's not buffer anymore as well)

Reply:
No, it's still buffer. Buffer is an array of bytes and instances of Uint8Array are buffers.

Regarding this line: `codePointAt` can return `undefined` there if `m` variable is an empty string. But this should be impossible here, as a callback iterates over symbols. I will add comment there to make it clearer.

Reply:
Sorry for not noticing the comment. In Node.js terms actually the reverse is true: https://nodejs.org/api/buffer.html#buffer
Buffer is a subclass of Uint8Array - basically this is the reason I suggested the change

Reply:
Makes sense i think. I can use different naming to separate it from node:buffer